ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClient.java

```diff
@@ -57,87 +59,77 @@ public Set<Relationship> getRelationships() {
-								"Starting url could not be determined for fragmentId: " + dataSourceUrl));
+	private RequestExecutor getRequestExecutor(final ProcessContext context) {
+		return switch (getAuthorizationStrategy(context)) {
+			case NO_AUTH -> requestExecutorFactory.createNoAuthExecutor();
+			case API_KEY -> requestExecutorFactory.createApiKeyExecutor(getApiKey(context), getApiKeyHeader(context));
+			case OAUTH2_CLIENT_CREDENTIALS ->
+				requestExecutorFactory.createClientCredentialsExecutor(getOauthClientId(context),
+						getOauthClientSecret(context), getOauthTokenEndpoint(context), getOauthScope(context));
+		};
+	}
+
+	private void determineLdesProperties(Ldes ldes, RequestExecutor requestExecutor, ProcessContext context) {
+		boolean timestampPath = streamTimestampPathProperty(context);
+		boolean versionOfPath = streamVersionOfProperty(context);
+		boolean shape = streamShapeProperty(context);
+		ldesProperties = new LdesPropertiesExtractor(requestExecutor).getLdesProperties(ldes, timestampPath,
+				versionOfPath, shape);
 	}
 
 	@Override
 	public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
-		if (ldesService.hasFragmentsToProcess()) {
-			Lang dataDestinationFormat = LdesProcessorProperties.getDataDestinationFormat(context);
-			LdesFragment fragment = ldesService.processNextFragment();
-
-			String timestampPath = streamTimestampPathProperty(context) ? fragment.getTimestampPath()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.LDES_TIMESTAMP_PATH.toString()))
-					: null;
-			String versionOfPath = streamVersionOfProperty(context) ? fragment.getVersionOfPath()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.LDES_VERSION_OF.toString()))
-					: null;
-			String shape = streamShapeProperty(context) ? fragment.getShaclShape()
-					.orElseThrow(() -> new LdesPropertyNotFoundException(LdesFragment.TREE_SHAPE.toString()))
-					: null;
-
-			// Send the processed members to the next Nifi processor
-			fragment.getMembers()
-					.forEach(ldesMember -> {
-						FlowFile flowFile = session.create();
-
-						if (streamTimestampPathProperty(context)) {
-							session.putAttribute(flowFile, "ldes.timestamppath", timestampPath);
-						}
-						if (streamVersionOfProperty(context)) {
-							session.putAttribute(flowFile, "ldes.isversionofpath", versionOfPath);
-						}
-						if (streamShapeProperty(context)) {
-							session.putAttribute(flowFile, "ldes.shacleshapes", shape);
-						}
-
-						FlowManager.sendRDFToRelation(session, flowFile,
-								ModelConverter.convertModelToString(ldesMember.getMemberModel(), dataDestinationFormat),
-								DATA_RELATIONSHIP, dataDestinationFormat);
-					});
+		SuppliedMember memberRecord = memberSupplier.get();
+
+		FlowFile flowFile = session.create();
+
+		if (streamTimestampPathProperty(context)) {
+			session.putAttribute(flowFile, "ldes.timestamppath", ldesProperties.getTimestampPath());
```

Comment:
does this need to be put on the session or flowfile? I'd think the session would keep on providing this value?

Reply:
Since the line above (105) says session.create(), I assume nothing is preserved.